src/Symfony/Bridge/PhpAmqp/AmqpSender.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpAmqp;
+
+use Symfony\Component\Messenger\Transport\SenderInterface;
+use Symfony\Component\Messenger\Transport\Serialization\EncoderInterface;
+
+/**
+ * Symfony Message sender to send messages to AMQP brokers using PHP's AMQP extension.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class AmqpSender implements SenderInterface
+{
+    private $messageEncoder;
+    private $connection;
+
+    public function __construct(EncoderInterface $messageEncoder, Connection $connection)
+    {
+        $this->messageEncoder = $messageEncoder;
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send($message)
+    {
+        $encodedMessage = $this->messageEncoder->encode($message);
+
+        $this->connection->publish(
```

Comment:
I'm not sure it's conceptually correct to publish to a **connection**.
I would expect to see `exchange->publish()` here (same goes for the queue).

I would have inverted the dependency, by exposing a `new Queue(new Connection)`, not the contrary.

Maybe you wrapped all the API in a `Connection` for ease of use ?